Return an Option<GroupMember>

SheaFoxx opened the pull request:
User might not be a member of the group.

<!-- Checklist:
- Format code
- Write proper PR title & description, add tags & so on.
- Check the issues and add "Closes #x" to end of description if applicable. -->

Commit:
Return an Option<GroupMember>

User might not be a member of the group.

l-2-j reviewed (approved):
Hmh, I thought that it'd just e fine to return an error when user is not a member...? ... I guess the semantics of an error vs an expected `None` are a bit different...

Commit:
Update groups.rs

target_id can be an instance id when creating an instance, or vrc_admin for vote kicks.

Commit:
Update groups.rs

Sorry, actor_id is vrc_admin for vote kicks, not target_id.

Commit:
Update groups.rs

They cannot be optional

Commit:
Update groups.rs

Serde with still required

Commit:
Update groups.rs

target_id can be none for group join events
